src/gfx/rgba.cpp

```diff
@@ -58,7 +58,15 @@ uint16_t Rgba::cgbColor() const {
 
 uint8_t Rgba::grayIndex() const {
 	assume(isGray());
-	// Convert from 0..<256 to hasTransparentPixels..<nbColorsPerPal
+	// 2bpp shades are inverted from RGB PNG; %00 = white, %11 = black
+	uint8_t gray = 255 - red;
+	if (options.palSpecType == Options::DMG) {
+		assume(!options.hasTransparentPixels);
+		// Reduce gray shade from 0..<256 to 0..<4, then map to color index,
+		// then reduce to 0..<nbColorsPerPal
+		return options.dmgColors[gray * 4 / 256] * options.nbColorsPerPal / 4;
```

Comment:
This kind of calculation should normally be `gray / 64` directly (or `gray / (256 / 4)` knowing that the result is an integer).

In this case, this doesn't matter. The value will expand, because `gray` is `uint8_t`, so it will be converted to `int` for the calculation. But in general, doing the `* 4` first can overflow. It's usually best to reduce these calculations as a general rule, because in other contexts they _can_ make a difference.

Reply:
(Apparently my own code has `rgb * 33 / 4` and a few `margin = dim * 3 / 10`s. No overflow issues in their ranges either though.)